waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/DefaultMetaStoreClientFactory.java

```diff
@@ -77,29 +77,40 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
         String user = (String) args[0];
         List<String> groups = (List<String>) args[1];
         cachedUgi = new HiveUgiArgs(user, groups);
-        return Lists.newArrayList(user);
+        if (base.isOpen()) {
```

Comment:
this should make sure that if setUgi is called on an open connect we pass it along. (From Hive CLI client this does not happen but perhaps other clients (Trino) it does.